pkg/bidstrategy/type.go

```diff
@@ -25,6 +25,14 @@ func NewShouldBidResponse() BidStrategyResponse {
 	}
 }
 
+type SemanticBidStrategy interface {
+	ShouldBid(ctx context.Context, request BidStrategyRequest) (BidStrategyResponse, error)
+}
+
+type ResourceBidStrategy interface {
+	ShouldBidBasedOnUsage(ctx context.Context, request BidStrategyRequest, usage model.ResourceUsageData) (BidStrategyResponse, error)
+}
+
```

Comment:
Double plus good change. Can we redefine the below `BidStrategy` to just be a combination of these two interfaces?

Reply:
Yup, done in latest commit.